### PR TITLE
Update webpacker to 5.x

### PR DESCRIPTION
**Pull request by stenington**
This PR upgrades webpacker to 5.1.1, which hopefully resolves some security alerts, or lets us upgrade other libraries that will resolve security alerts.

The `CommonsChunkPlugin` has been replaced with `splitChunks` configuration as detailed [here](https://github.com/rails/webpacker/blob/master/docs/v4-upgrade.md#splitchunks-configuration).

A CSP initializer has been added in development to let `webpack-dev-server` work, as described [here](https://github.com/rails/webpacker#development).

Much of the config you see in this PR was auto-generated by webpacker; we can always customize it further in the future if/when the need arises.

The test suite now outputs a handful of deprecation notices as well, but I think that's better dealt with separately.

### CODE REVIEW

File: app/javascript/packs/admin.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import "core-js/stable";
+import "regenerator-runtime/runtime";
+
```
Comment (stenington) [inline app/javascript/packs/admin.js:3]: I believe the above two statements are supposed to be put at the top of each pack, and replace the `babel-polyfill` import some packs made.

File: app/views/layouts/admin.html.erb [hunk 2]
```diff
@@ -11,14 +11,14 @@
 
     <%= stylesheet_link_tag 'admin', media: 'all' %>
 
-    <%= stylesheet_pack_tag "vendor" %>
+    <%# stylesheet_pack_tag "vendor" %>
```
Comment (stenington) [inline app/views/layouts/admin.html.erb:14]: The vendor and manifest packs came from the `CommonsChunkPlugin`, which is replaced in webpack v4 with `SplitChunks` configuration. ([see here](https://github.com/rails/webpacker/blob/master/docs/v4-upgrade.md#splitchunks-configuration)) For now I'm forgoing this optimization and just trying to get things working without splitting out chunks.

File: babel.config.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+module.exports = function(api) {
```
Comment (stenington) [inline babel.config.js:1]: This was generated by `rails webpacker:install`, and I think it replaces the `.babelrc`. I have no idea if that one was also auto-generated, or has customizations we might want to port.

File: config/webpack/environment.js [hunk 4]
```diff
@@ -1,44 +1,10 @@
...
 const vue = require('./loaders/vue')
 
-environment.loaders.append('vue', vue)
-
+const customConfig = require('./custom')
 environment.config.merge(customConfig)
 
-const webpack = require('webpack')
-const dotenv = require('dotenv');
-
```
Comment (stenington) [inline config/webpack/environment.js:11]: Below are the `CommonsChunkPlugin` bits that need porting to `environment.splitChunks` I think.

File: config/webpack/environment.js [hunk 5]
```diff
@@ -1,44 +1,10 @@
...
-  })
-)
-
-environment.plugins.prepend(
-  'MomentIgnoreLocales',
-  new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/)
-)
-
-// Pull in .env variables used by front-end
-dotenv.config({ path: '.env', silent: true })
```
Comment (stenington) [inline config/webpack/environment.js:37]: Probably need to bring this back.

File: package.json [hunk 6]
```diff
@@ -1,82 +1,33 @@
 {
-  "scripts": {
-    "test": "jest"
```
Comment (stenington) [inline package.json:3]: Need to bring back all the jest stuff.

File: postcss.config.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (stenington) [inline postcss.config.js:1]: This was also auto-generated and replaces `.postcssrc.yml` which again may have had customizations we need to port, I'm not sure.

**Comment by stenington** [comment]
At this stage of the PR, the javascript packs build and seem to work in some places, but not everywhere. A handful of rspec tests fail:

```
rspec ./spec/system/student/signup_spec.rb:5 # Students signing up authenticating logs in okay with the email / password used during signup
rspec ./spec/system/student/signup_spec.rb:51 # Students signing up location form lets you change its guess
rspec ./spec/system/student/create_team_spec.rb:5 # Students creating a team default team location lets you change the signup form default and sets the team to your change
rspec ./spec/system/mentor/signup_spec.rb:5 # Mentors signing up authenticating logs in okay with the email / password used during signup
rspec ./spec/system/mentor/complete_training_spec.rb:17 # A mentor completing their training is not required for new mentors before the ImportantDates.mentor_training_required_since date
rspec ./spec/system/mentor/complete_training_spec.rb:63 # A mentor completing their training is required on and after the ImportantDates.mentor_training_required_since date
```

and some selectors are broken during signup (possibly from upgrading `vue-select`:

Screen Shot 2020-07-24 at 4 16 56 PM

File: package.json [hunk 8]
```diff
@@ -1,82 +1,33 @@
...
-    "babel-polyfill": "^6.26.0",
-    "babel-preset-env": "^1.6.1",
+    "@rails/webpacker": "5.1.1",
+    "airbrake-js": "^1.4.0",
+    "axios": "^0.19.2",
+    "axios-case-converter": "^0.6.0",
     "chart.js": "^2.7.2",
     "chroma-js": "^1.3.7",
-    "dotenv": "^6.2.0",
-    "element-ui": "^2.2.0",
```
Comment (stenington) [inline package.json:44]: I'm unsure which, if any, of the missing dependencies need to come back.

**Comment by stenington** [comment]
@shaunxp20 As part of either code review or testing, can you confirm that the `webpack-dev-server` works on your machine? For me, I ran the rails development server in one terminal and `bin/webpack-dev-server` in another, and then javascript components would hot-load as I changed them.

**Comment by shaunxp20** [comment]
> @shaunxp20 As part of either code review or testing, can you confirm that the `webpack-dev-server` works on your machine? For me, I ran the rails development server in one terminal and `bin/webpack-dev-server` in another, and then javascript components would hot-load as I changed them.

I checked out this branch and verified that the `webpack-dev-server` works for me.

**Pull request merged**